kimchi/src/circuits/polynomials/foreign_field_add/witness.rs

```diff
@@ -267,90 +235,49 @@ fn init_foreign_field_fin_rows<F: PrimeField>(
 ) {
     let out_row = 8 * num; // row where the final result is stored in RC
     let bound_row = 8 * num + 4; // row where the final bound is stored in RC
-    let witness_shape: [[WitnessCell<F>; COLUMNS]; 2] = [
+    let witness_shape: [[Box<dyn WitnessCell<F>>; COLUMNS]; 2] = [
```

Comment:
Nit: these can probably be `&dyn WitnessCell<F>` to avoid the heap allocations.

Reply:
Hm.. Yeah -- I agree. I originally had it the way you suggest b/c it was my preferred way, but it was not happy to allocate them and was complaining about static lifetimes.  So, in interest of time, I gave up.  I will try to fix in a subsequent PR.